### PR TITLE
feat(snowflake-usage): add knob for direct objects accesssed vs base objects accessed

### PR DESCRIPTION
**Pull request by gabe-lyons**
`direct_objects_accessed` will display the view in the case a view is used. `base_objects_accessed` will display the underlying table the view was built off of. For usage, it makes more sense to default to `direct_objects_accessed`. This also includes a knob to use `base_objects_accessed` in the case a user would prefer that.

## Checklist
- [ ] The PR conforms to DataHub's [Contributing Guideline](https://github.com/linkedin/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/linkedin/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [ ] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable)

### CODE REVIEW

File: metadata-ingestion/source_docs/snowflake.md [hunk 1]
```diff
@@ -139,18 +139,19 @@ sink:
...
+| `env`                         |          | `"PROD"`                                                       | Environment to use in namespace when constructing URNs.         |
+| `bucket_duration`             |          | `"DAY"`                                                        | Duration to bucket usage events by. Can be `"DAY"` or `"HOUR"`. |
+| `start_time`                  |          | Last full day in UTC (or hour, depending on `bucket_duration`) | Earliest date of usage logs to consider.                        |
+| `end_time`                    |          | Last full day in UTC (or hour, depending on `bucket_duration`) | Latest date of usage logs to consider.                          |
+| `top_n_queries`               |          | `10`                                                           | Number of top queries to save to each table.                    |
+| `use_base_objects_accessed`   |          | False                                                          | Applies queries issued to views to the underlying table.        |
```
Comment (shirshanka) [inline metadata-ingestion/source_docs/snowflake.md:154]: s/Applies/Attributes

**Review by shirshanka — APPROVED** [review]
LGTM. Will wait for CI to be green

**Comment by shirshanka** [comment]
merged since cypress failure is unrelated.

File: metadata-ingestion/src/datahub/ingestion/source/usage/snowflake_usage.py [hunk 2]
```diff
@@ -195,7 +202,13 @@ def _aggregate_access_events(
                 event.query_start_time, self.config.bucket_duration
             )
 
-            for object in event.base_objects_accessed:
+            accessed_data = []
```
Comment (rslanka) [inline metadata-ingestion/src/datahub/ingestion/source/usage/snowflake_usage.py:205]:
```python
            accessed_data = event.base_objects_accessed  if self.config.apply_view_usage_to_tables else event.direct_objects_accessed```

**Pull request merged**